src/stable_marriage.rs

```diff
@@ -0,0 +1,164 @@
+struct Individual {
+    cnt_match: Option<usize>,
+
+    // For efficiency: most preferred last for men, most preferred first for women
+    is_man: bool,
+    prefs: Vec<usize>,
+}
+
+impl Individual {
+    fn get_pref_idx(&self, idx: usize) -> Option<usize> {
+        if self.is_man {
+            self.prefs.iter().rev().position(|i| *i == idx)
+        } else {
+            self.prefs.iter().position(|i| *i == idx)
+        }
+    }
+
+    fn pop_preferred_if_unmatched(&mut self) -> Option<usize> {
```

Comment:
This could deserve a `debug_assert!(self.is_man)`, in order to avoid confusing the reader :)